### PR TITLE
Documentation for Kubelet Configuration

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
Documentation for Kubelet Configuration

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/getting-started/optional/kubeletconfig.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Note that this is the preferred and supported way to specify any Kubelet settings from the release `v0.20.0` onwards. Previously the [`hostOSConfiguration.bottlerocketConfiguration.kubernetes`](https://anywhere.eks.amazonaws.com/docs/getting-started/optional/hostosconfig/#kubernetes) field was used to specify Bottlerocket Kubernetes settings. That has been deprecated from `v0.20.0`
+
+## Special fields
+
+### Duplicate fields
+
+The `clusterNetwork.dns.resolvConf` is the file path to a file containing a custom DNS resolver configuration. This can now be provided in the Kubelet Configuration using the `resolvConf` field. Note that if both these fields are set, the Kubelet Configuration's field will take precendence and override the value from the `clusterNetwork.dns.resolvConf`.
+
```
Comment (cxbrowne1207) [inline docs/content/en/docs/getting-started/optional/kubeletconfig.md:61]:
```suggestion
```

File: docs/content/en/docs/getting-started/optional/kubeletconfig.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+|                     | vSphere | Bare Metal | Nutanix | CloudStack | Snow |
+|:-------------------:|:-------:|:----------:|:-------:|:----------:|:----:|
+|    Ubuntu 20.04     |    ✔    |     ✔      |    ✔    |     —      |  —   |
+|    Ubuntu 22.04     |    ✔    |     ✔      |    ✔    |     —      |  —   |
+| Bottlerocket        |    ✔    |     —      |    —    |     —      |  —   |
+|      RHEL 8.x       |    ✔    |     ✔      |    ✔    |     ✔      |  —   |
+|      RHEL 9.x       |    —    |     —      |    ✔    |     ✔      |  —   |
+
+
+You can configure EKS Anywhere to specify Kubelet settings and configure those for control plane and/or worker nodes starting from `v0.20.0`. This can be done using `kubeletConfiguration`.
```
Comment (cxbrowne1207) [inline docs/content/en/docs/getting-started/optional/kubeletconfig.md:23]: We now have versioned docs, so do we need to explicitly mention the version here?

Comment (mitalipaygude) [reply docs/content/en/docs/getting-started/optional/kubeletconfig.md:23]: Yeah I couldn't find the specific v0.20.0 docs on our docs page so I added this extra info for now, can probably remove it later on.

File: docs/content/en/docs/getting-started/optional/kubeletconfig.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## Special fields
+
+### Duplicate fields
+
+The `clusterNetwork.dns.resolvConf` is the file path to a file containing a custom DNS resolver configuration. This can now be provided in the Kubelet Configuration using the `resolvConf` field. Note that if both these fields are set, the Kubelet Configuration's field will take precendence and override the value from the `clusterNetwork.dns.resolvConf`.
+
+
+### Blocked fields
+
+Fields like `providerID` or `cloudProvider` are set by EKS Anywhere and can't  be set by users. This is to maintain seamless support for all providers.
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/optional/kubeletconfig.md:65]:
```suggestion
Fields like `providerID` or `cloudProvider` are set by EKS Anywhere and can't be set by users. This is to maintain seamless support for all providers.
```

File: docs/content/en/docs/getting-started/optional/kubeletconfig.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+More details on the Kubelet Configuration object and its supported fields can be found [here](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/). EKS Anywhere only supports the latest Kubernetes version's `KubeletConfiguration`.
+
+## Bottlerocket Support
+
+The only provider that supports `kubeletConfiguration` with Bottlerocket is vSphere. The list of settings that can be configured for Bottlerocket can be found [here](https://bottlerocket.dev/en/os/1.19.x/api/settings/kubernetes/#alphaorder). This page also describes other settings for various settings like Kubelet Options. The settings supported by Bottlerocket will have information specific to the `Kubelet Configuration` keyword in there. Refer to the documentation to learn about the supported fields as well as their data types as they may vary from the upstream object's data types.
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/optional/kubeletconfig.md:52]:
```suggestion
The only provider that supports `kubeletConfiguration` with Bottlerocket is vSphere. The list of settings that can be configured for Bottlerocket can be found [here](https://bottlerocket.dev/en/os/1.19.x/api/settings/kubernetes/#alphaorder). This page also describes other various settings, like Kubelet Options. The settings supported by Bottlerocket will have information specific to the `Kubelet Configuration` keyword in there. Refer to the documentation to learn about the supported fields as well as their data types as they may vary from the upstream object's data types.
```

**Review by chrisnegus — COMMENTED** [review]
Looks good. A couple of tiny suggestions.

**Pull request merged**